rusoto/core/src/signature.rs

```diff
@@ -391,24 +400,34 @@ fn to_hexdigest<T: AsRef<[u8]>>(t: T) -> String {
     h.as_ref().to_hex().to_string()
 }
 
+fn build_scheme(region: Region) -> String {
```

Comment:
Consider just inlining this function, it's simple enough and only used in one location. Matching over `(self.scheme, self.region)` could simplify it.